lua/auto-hlsearch.lua

```diff
@@ -1,23 +1,39 @@
+local M = {}
+
 local defaults = {
   remap_keys = { "/", "?", "*", "#", "n", "N" },
 }
 
--- Remap provided keys in order to use :AutoHlsearch command
+-- Remap provided keys in order to use activate() function
 -- while keeping the user's keymap configuration
 local function remap_keys(keys)
   local function set(lhs, keymap)
-    local cmd = ":AutoHlsearch<CR>"
-    if keymap and keymap.rhs then
+    if keymap then
       local opts = {
-        expr = keymap.expr,
+        expr = true,
         noremap = keymap.noremap,
         nowait = keymap.nowait,
         script = keymap.script,
         silent = keymap.silent,
       }
-      vim.api.nvim_set_keymap("n", lhs, string.format("%s%s", cmd, keymap.rhs), opts)
+      -- We need to consider the remmaping when use expr options
+      if keymap.expr == 1 then
```

Comment:
why do we need to check it?
if the user has a lua function in a keymap, it doesn't matter if `keymap.expr == 1` or not, the function should be executed anyway. am I right?

Reply:
I see. I have refactored it.

Reply:
Nice, looks much better!